persistence/javadsl/src/main/java/com/lightbend/lagom/javadsl/persistence/ChildPersistentEntity.java

```diff
@@ -0,0 +1,115 @@
+import scala.compat.java8.FutureConverters;
+import scala.compat.java8.OptionConverters;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+import scala.runtime.AbstractFunction0;
+
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A persistent entity that runs as a child actor of the current actor context.
+ */
+public class ChildPersistentEntity<Command> {
+
+  private final ActorRef actor;
+
+  ChildPersistentEntity(ActorRef actor) {
+    this.actor = actor;
+  }
+
+  /**
+   * Get the actual actor for the entity.
+   * <p>
+   * This may be useful in order to watch the actors termination.
+   */
+  public ActorRef getActor() {
+    return actor;
+  }
+
+  /**
+   * Send a message to the actor.
+   */
+  public void tell(Command msg, ActorRef sender) {
+    actor.tell(msg, sender);
+  }
+
+  /**
+   * Ask the actor with a command.
+   */
+  @SuppressWarnings("unchecked")
+  public <Reply, Cmd extends PersistentEntity.ReplyType<Reply>> CompletionStage<Reply> ask(Cmd msg, Timeout timeout) {
+    return FutureConverters.toJava((Future<Reply>) Patterns.ask(actor, msg, timeout));
+  }
+
+  /**
+   * Forward a message to the actor.
+   */
+  public void forward(Command msg, ActorContext ctx) {
+    actor.forward(msg, ctx);
+  }
+
+  /**
+   * Stop the persistent entity.
+   */
+  public void stop() {
+    actor.tell(PersistentEntityActor.Stop$.MODULE$, ActorRef.noSender());
+  }
+
+  /**
+   * Instantiate a persistent entity as a child of the current actor context.
+   * <p>
+   * Note that since this is a direct child, this provides no guarantee that this is the only entity with the given id
+   * running in the cluster, or even in the actor system. It is up to the actor that instantiates this to ensure that
+   * only one instance of the entity across the cluster is instantiated (by using cluster sharding to distribute
+   * itself, for example), or to deal with inconsistencies that may arise due to having multiple entities instantiated
+   * in the cluster.
+   *
+   * @param entityClass The class of the persistent entity.
+   * @param injector    The injector to use to create the persistent entity.
+   * @param entityId    The entity ID.
+   * @param actorName   The actor name.
+   * @param ctx         The actor context.
+   */
+  @SuppressWarnings("unchecked")
+  public static <Command> ChildPersistentEntity<Command> create(Class<? extends PersistentEntity<Command, ?, ?>> entityClass,
+      Injector injector, String entityId, String actorName, ActorContext ctx) {
+
+    Optional<ActorRef> maybeEntity = OptionConverters.toJava(ctx.child(actorName));
+
+    ActorRef actorRef = maybeEntity.orElseGet(() -> {
+      Config conf = ctx.system().settings().config().getConfig("lagom.persistence");
+      Optional<Object> snapshotAfter = Optional.of(conf.getString("snapshot-after"))
+          .filter(str -> !str.equals("off"))
+          .map(str -> conf.getInt("snapshot-after"));
```

Comment:
This code to parse `snapshot-after` is duplicate between this class and `AbstractPersistentEntityRegistry` (both for java and scala). I think we could simplify to have at most one version of this code for java and one for scala (though a single copy for all 4 occurrences would be great).